scripts/sync_tensorflow_protobuf_stubs.sh

```diff
@@ -3,63 +3,70 @@ set -euxo pipefail
 
-mkdir repository
+cd "$(dirname "$0")" > /dev/null
+cd ../stubs/tensorflow
+mkdir -p repository
 pushd repository &> /dev/null
-    git clone https://github.com/tensorflow/tensorflow.git
+    # If the script fails halfway, it's nice to be able to re-run it immediately
+    if [ ! -d "tensorflow" ] ; then
+        git clone --depth 1 --branch v"$TENSORFLOW_VERSION" https://github.com/tensorflow/tensorflow.git
+    fi
     pushd tensorflow &> /dev/null
-        git checkout v"$TENSORFLOW_VERSION"
-
         # Folders here cover the more commonly used protobufs externally and
         # their dependencies. Tensorflow has more protobufs and can be added if requested.
         protoc --mypy_out "relax_strict_optional_primitives:$REPO_ROOT/stubs/tensorflow" \
+            tensorflow/compiler/xla/*.proto \
+            tensorflow/compiler/xla/service/*.proto \
+            tensorflow/core/example/*.proto \
+            tensorflow/core/framework/*.proto \
             tensorflow/core/protobuf/*.proto \
             tensorflow/core/protobuf/tpu/*.proto \
-            tensorflow/core/framework/*.proto \
             tensorflow/core/util/*.proto \
-            tensorflow/core/example/*.proto \
             tensorflow/python/keras/protobuf/*.proto \
-            tensorflow/tsl/protobuf/*.proto \
-            tensorflow/compiler/xla/*.proto \
-            tensorflow/compiler/xla/service/*.proto
+            tensorflow/tsl/protobuf/*.proto
     popd &> /dev/null
 popd &> /dev/null
 
-rm -rf repository/
 # These protos exist in a folder with protos used in python, but are not
 # included in the python wheel. They are likely only used for other
 # language builds. stubtest was used to identify them by looking for
 # ModuleNotFoundError.
-rm tensorflow/core/protobuf/coordination_service_pb2.pyi \
-   tensorflow/compiler/xla/service/hlo_execution_profile_data_pb2.pyi \
+rm tensorflow/compiler/xla/service/hlo_execution_profile_data_pb2.pyi \
    tensorflow/compiler/xla/service/hlo_profile_printer_data_pb2.pyi \
    tensorflow/compiler/xla/service/test_compilation_environment_pb2.pyi \
+   tensorflow/compiler/xla/xla_pb2.pyi \
    tensorflow/core/protobuf/autotuning_pb2.pyi \
    tensorflow/core/protobuf/conv_autotuning_pb2.pyi \
    tensorflow/core/protobuf/critical_section_pb2.pyi \
    tensorflow/core/protobuf/eager_service_pb2.pyi \
    tensorflow/core/protobuf/master_pb2.pyi \
    tensorflow/core/protobuf/master_service_pb2.pyi \
    tensorflow/core/protobuf/replay_log_pb2.pyi \
+   tensorflow/core/protobuf/tpu/compile_metadata_pb2.pyi \
    tensorflow/core/protobuf/worker_pb2.pyi \
    tensorflow/core/protobuf/worker_service_pb2.pyi \
-   tensorflow/core/protobuf/tpu/compile_metadata_pb2.pyi \
-   tensorflow/core/util/example_proto_fast_parsing_test_pb2.pyi \
-   tensorflow/compiler/xla/xla_pb2.pyi
+   tensorflow/core/util/example_proto_fast_parsing_test_pb2.pyi
+
+isort "$REPO_ROOT/stubs/tensorflow/tensorflow"
```

Comment:
If #10912 is merged first, this becomes:
```suggestion
ruff "$REPO_ROOT/stubs/tensorflow/tensorflow" --exit-non-zero-on-fix --fix-only
```